src/server/rest/v1/service/SiteAreaService.ts

```diff
@@ -354,26 +363,37 @@ export default class SiteAreaService {
         }
       }
     }
-    siteArea.numberOfPhases = filteredRequest.numberOfPhases;
-    if (Utils.isComponentActiveFromToken(req.user, TenantComponents.OCPI)) {
-      if (Utils.objectHasProperty(filteredRequest, 'tariffID')) {
-        siteArea.tariffID = filteredRequest.tariffID;
-      }
-      if (!site.public) {
-        delete siteArea.tariffID;
-      }
+    if (siteArea.siteID !== filteredRequest.siteID) {
+      additionalSiteIDs.push(siteArea.siteID);
```

Comment:
@LucasBrazi06 Normally this should be the case already. The end user has to detach the site area first if he wants to change sites.
The reason for the array is that if you change the siteID of a parent, you also need to check if there are children left on the old site. So with the additional siteID, the trees from both sites (old and new) are build and checked for errors. If there is a child with another site id, an error is thrown. A better name for the array would probably be additionalSitesIDsToCheck.

We could also use the dependency method to check this, but this would mean another request to the db.

Reply:
@LucasBrazi06 Normally this should be the case already. The end user has to detach the site area first if he wants to change sites.
The reason for the array is that if you change the siteID of a parent, you also need to check if there are children left on the old site. So with the additional siteID, the trees from both sites (old and new) are build and checked for errors. If there is a child with another site id, an error is thrown. A better name for the array would probably be additionalSitesIDsToCheck.

We could also use the dependency method to check this, but this would mean another request to the db.

Reply:
> @LucasBrazi06 Normally this should be the case already. The end user has to detach the site area first if he wants to change sites. The reason for the array is that if you change the siteID of a parent, you also need to check if there are children left on the old site. So with the additional siteID, the trees from both sites (old and new) are build and checked for errors. If there is a child with another site id, an error is thrown. A better name for the array would probably be additionalSitesIDsToCheck.
> 
> We could also use the dependency method to check this, but this would mean another request to the db.

This is what I wrote, I think we should not allow to change site unless it's not part of the tree.
The tree should contain only Site Areas within the same Site always.

Reply:
@LucasBrazi06 Yes this is how it has been handled since the beginning.
Let me explain in another way maybe. 
Let us assume we have an existing tree. Now someone wants to change the site of the parent. Then we also need to check if there are dependencies on the old site (existing children). And this is what additionalSites is doing.
Otherwise we would only check the new one, and there are no regressions of course. But on the old one we have children without a parent:
![image](https://user-images.githubusercontent.com/33114317/158142220-5350337d-5b46-47b6-b4cf-4d2e294cc8cf.png)

Reply:
I can change the way of checking and use the dependency method.